source/frontends/sdl/imgui/sdldebugger.cpp

```diff
@@ -373,24 +372,61 @@ namespace sa2
           if (bDisasmFormatFlags & DISASM_FORMAT_TARGET_POINTER)
           {
             debuggerTextColored(FG_DISASM_ADDRESS, line.sTargetPointer);
+            
+            if (bDisasmFormatFlags & DISASM_FORMAT_TARGET_VALUE)
+            {
+              ImGui::SameLine(0, 0);
+              debuggerTextColored(FG_DISASM_OPERATOR, ":");
+              ImGui::SameLine(0, 0);
+              debuggerTextColored(FG_DISASM_OPCODE, line.sTargetValue);
+            }
           }
-
-          ImGui::TableNextColumn();
-          if (bDisasmFormatFlags & DISASM_FORMAT_TARGET_VALUE)
+          else if (line.bTargetImmediate && line.nImmediate)
           {
-            debuggerTextColored(FG_DISASM_OPCODE, line.sTargetValue);
+            debuggerTextColored(FG_DISASM_OPERATOR, "#");
+            ImGui::SameLine(0, 0);
+            debuggerTextColored(FG_DISASM_SINT8, line.sImmediateSignedDec);
           }
 
           ImGui::TableNextColumn();
-          if (bDisasmFormatFlags & DISASM_FORMAT_CHAR)
+          if (bDisasmFormatFlags & DISASM_FORMAT_BRANCH)
           {
-            ImGui::TextUnformatted(line.sImmediate);
+            if ((unsigned char)*line.sBranch == 0x8A)
+            {
+              debuggerTextColored(FG_DISASM_BRANCH, "v");
+            }
+            else if ((unsigned char)*line.sBranch == 0x8B)
+            {
+              debuggerTextColored(FG_DISASM_BRANCH, "^");
+            }
+            else
+            {
+              ImGui::TextUnformatted(line.sBranch);
+            }
           }
 
           ImGui::TableNextColumn();
-          if (bDisasmFormatFlags & DISASM_FORMAT_BRANCH)
+          if (bDisasmFormatFlags & DISASM_FORMAT_CHAR)
           {
-            ImGui::TextUnformatted(line.sBranch);
+            DebugColors_e color;
+            unsigned int c = (unsigned char)line.nImmediate;
+            if ((c & 0x7F) < 0x20)
```

Comment:
is this matching 0x00..0x1f and 0x80..0x9f? I forgot how it works, but I would add a comment.

Reply:
added comments